realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -557,6 +582,19 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
         }
     }
 
+    /**
+     * Trigger onSuccess for all completed async transaction.
+     * NOTE: Should only be called from {@link #notifyAllListeners(List)}.
+     */
+    private void notifyAsyncTransactionCallbacks() {
+        if (!pendingOnSuccessAsyncTransactionCallbacks.isEmpty()) {
+            for (Runnable callback : pendingOnSuccessAsyncTransactionCallbacks) {
+                realm.handler.post(callback);
```

Comment:
I feel something will go wrong by posting the runnable instead of calling it directly. Consider by posting it, other threads will have chance to post a change event in front of it. What is the reason do posting here?